src/ISA/ISA/JsonTypes/OntologyAnnotation.fs

```diff
@@ -58,137 +59,79 @@ type OntologyAnnotation =
+        let regexResult = 
+            Regex.tryGetTermAccessionString termAccessionNumber
+
+        let nextTermAccessionNumber = regexResult |> Option.defaultValue termAccessionNumber
 
         OntologyAnnotation.make 
             None 
             (Option.fromValueWithDefault "" term |> Option.map AnnotationValue.fromString)
-            (Option.fromValueWithDefault "" source)
-            (Option.fromValueWithDefault "" accessionNumber)
+            (Option.fromValueWithDefault "" termSourceRef)
+            (Option.fromValueWithDefault "" nextTermAccessionNumber)
             None
 
-    /// Create a ISAJson Ontology Annotation value from ISATab string entries
+    /// Create a ISAJson Ontology Annotation value from string entries
     static member fromStringWithComments (term:string) (source:string) (accessionNumber:string) (comments : Comment list) =
-        
+
         OntologyAnnotation.make 
             None 
             (Option.fromValueWithDefault "" term |> Option.map AnnotationValue.fromString)
             (Option.fromValueWithDefault "" source)
-            (Option.fromValueWithDefault "" accessionNumber |> Option.map URI.fromString)
+            (Option.fromValueWithDefault "" accessionNumber)
             (Option.fromValueWithDefault [] comments)
 
-    static member fromAnnotationId (id : string) =
-        id 
-        |> OntologyAnnotation.splitAnnotation
-        |> fun (source,num) ->
-            OntologyAnnotation.fromString "" source id
-
-    member this.ShortAnnotationString = 
-        match this.TermAccessionNumber with
-        | Some t ->
-            match OntologyAnnotation.trySplitUri t with 
-            | Some (s,t) -> OntologyAnnotation.createShortAnnotation s t
-            | None -> 
-                if System.Text.RegularExpressions.Regex.Match(t,OntologyAnnotation.shortAnnotationRegex).Success then t
-                else ""
-        | None -> ""
-
-    member this.URLAnnotationString = 
-        match this.TermAccessionNumber with
-        | Some t ->
-            match OntologyAnnotation.trySplitShortAnnotation t with 
-            | Some (s,t) -> OntologyAnnotation.createUriAnnotation s t
-            | None -> 
-                let r = System.Text.RegularExpressions.Regex.Match(t,OntologyAnnotation.ontologyTermURIRegex)
-                if r.Success then t
-                else ""
+    /// Will always be created without `OntologyAnnotion.Name`
+    static member fromTermAccession (termAccession : string) =
+        termAccession
+        |> Regex.tryParseTermAccession
+        |> Option.get 
+        |> fun r ->
+            let accession = r.IdSpace + ":" + r.LocalId
+            OntologyAnnotation.fromString "" r.IdSpace accession
+
+    /// Parses any value in `TermAccessionString` to term accession format "idspace:localid". Exmp.: "MS:000001".
+    ///
+    /// If `TermAccessionString` cannot be parsed to this format, returns empty string!
+    member this.TermAccessionShort = 
```

Comment:
These functions should be really well tested. (Because of the different possibilities of input)